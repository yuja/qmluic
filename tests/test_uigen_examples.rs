pub mod common;

#[test]
fn test_gadget_properties() {
    insta::assert_snapshot!(common::translate_file("examples/GadgetProperties.qml").unwrap());
}

#[test]
fn test_item_views() {
    insta::assert_snapshot!(common::translate_file("examples/ItemViews.qml").unwrap());
}

#[test]
fn test_layout_flow() {
    insta::assert_snapshot!(common::translate_file("examples/LayoutFlow.qml").unwrap());
}

#[test]
fn test_main_window() {
    insta::assert_snapshot!(common::translate_file("examples/MainWindow.qml").unwrap());
}

#[test]
fn test_settings_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/SettingsDialog.qml").unwrap());
}

#[test]
fn test_static_item_model() {
    insta::assert_snapshot!(common::translate_file("examples/StaticItemModel.qml").unwrap());
}

#[test]
fn test_various_layouts() {
    insta::assert_snapshot!(common::translate_file("examples/VariousLayouts.qml").unwrap());
}

#[test]
fn test_customwidget_common_my_dialog_button_box() {
    insta::assert_snapshot!(common::translate_file(
        "examples/customwidget/common/MyDialogButtonBox.qml"
    )
    .unwrap());
}

#[test]
fn test_customwidget_main_dialog() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/MainDialog.qml").unwrap()
    );
}

#[test]
fn test_customwidget_settings_form() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/SettingsForm.qml").unwrap()
    );
}

#[test]
fn test_thg_hg_email_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/thg/HgEmailDialog.qml").unwrap());
}
