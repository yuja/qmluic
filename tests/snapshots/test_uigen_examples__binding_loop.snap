---
source: tests/test_uigen_examples.rs
expression: ui_xml
---
<ui version="4.0">
 <class>BindingLoop</class>
 <widget class="QWidget" name="widget">
  <layout class="QVBoxLayout" name="vboxLayout">
   <item>
    <widget class="QPushButton" name="trig">
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Click to crash</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item column="0" row="0">
      <widget class="QCheckBox" name="check0">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="1" row="0">
      <widget class="QCheckBox" name="check1">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="2" row="0">
      <widget class="QCheckBox" name="check2">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="3" row="0">
      <widget class="QCheckBox" name="check3">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="4" row="0">
      <widget class="QCheckBox" name="check4">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="5" row="0">
      <widget class="QCheckBox" name="check5">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="6" row="0">
      <widget class="QCheckBox" name="check6">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="7" row="0">
      <widget class="QCheckBox" name="check7">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="0" row="1">
      <widget class="QCheckBox" name="check8">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="1" row="1">
      <widget class="QCheckBox" name="check9">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="2" row="1">
      <widget class="QCheckBox" name="check10">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="3" row="1">
      <widget class="QCheckBox" name="check11">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="4" row="1">
      <widget class="QCheckBox" name="check12">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="5" row="1">
      <widget class="QCheckBox" name="check13">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="6" row="1">
      <widget class="QCheckBox" name="check14">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="7" row="1">
      <widget class="QCheckBox" name="check15">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="0" row="2">
      <widget class="QCheckBox" name="check16">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="1" row="2">
      <widget class="QCheckBox" name="check17">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="2" row="2">
      <widget class="QCheckBox" name="check18">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="3" row="2">
      <widget class="QCheckBox" name="check19">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="4" row="2">
      <widget class="QCheckBox" name="check20">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="5" row="2">
      <widget class="QCheckBox" name="check21">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="6" row="2">
      <widget class="QCheckBox" name="check22">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="7" row="2">
      <widget class="QCheckBox" name="check23">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="0" row="3">
      <widget class="QCheckBox" name="check24">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="1" row="3">
      <widget class="QCheckBox" name="check25">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="2" row="3">
      <widget class="QCheckBox" name="check26">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="3" row="3">
      <widget class="QCheckBox" name="check27">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="4" row="3">
      <widget class="QCheckBox" name="check28">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="5" row="3">
      <widget class="QCheckBox" name="check29">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="6" row="3">
      <widget class="QCheckBox" name="check30">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="7" row="3">
      <widget class="QCheckBox" name="check31">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item column="0" row="4">
      <widget class="QCheckBox" name="check32">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="spacerItem">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
</ui>

